### DISCUSSION TITLE
new build

### DISCUSSION THREAD

**Discussion by games133** [awawa-dev/HyperHDR#621] [General]
hi as title states im current building a system this is my first time with a raspberry pi or anything else like this. so far ive managed to install pi software and hyperhdr. as for the 3.3 - 5v i thought id get a adafruit feather and got the software on there. this is where im stuck not 100% sure how to connect them together. also need v20 of hdr. ive looked arougnt his site and not had much look finding v20 i know its in beta

any help would be much appreciated. ideally id like some pics of a system with adafruit feather and pi 4 connected.  and possible direct link to v20

thanks in advance

**Comment by awawa-dev** [comment]
Hi
You can find v20beta0 installers inside archives/Artifacts here: https://github.com/awawa-dev/HyperHDR/actions/runs/5825821246 (latest Github Action, will be removed after some time)

**Reply by awawa-dev** [reply]
Currently bullseye version is valid for Raspberry Pi OS. 32 vs 64 version depending on the system you have already installed.

**Reply by games133** [reply]
ok so really dumb question and complete idote how do you download. pressing on either the 2 32 and 64 and cant see download option 

sorry for been a pain

**Reply by awawa-dev** [reply]
You must download one of these and extract it:
![obraz](https://github.com/awawa-dev/HyperHDR/assets/69086569/5d55123d-26cf-40a8-87f3-91d06a5c7fb3)

**Reply by games133** [reply]
aaahhhh ha found it thanks very much. 

just need so how connect the pi to the adafruit board now.

**Comment by games133** [comment]
dose anyone have circut digam on how to connect the rasberry pi4 to the adafruit the that to the light something simple?

**Reply by awawa-dev** [reply]
What specific version of 'adafruit feather' board (there are a few of them) are you using and what exactly firmware/file (from where? link) did you upload to it?

**Reply by games133** [reply]
ive got adafruit feather rp2040 scorpio.  i put hyperserialpico_sk6812neutral.uf2 on it from :- https://github.com/awawa-dev/HyperSerialPico/releases

ive relised its pin 0 on the scorpio for data to lights. but not to sure how to connect the raspberry to it. 

so far ive got 5v 20 amp power supply
raspberry pi4 4gb ram with 64bit softwear. and hyper hdr v20 beta (thanks to you. couldnt belive how stupid i was not to see it)
feintech hdmi splitter
ezcoo hdmi splitter. 
4k capture car https://www.amazon.co.uk/Capture-Record-Camcorder-Action-Switch/dp/B0BK93341Y/ref=sr_1_2_sspa?keywords=4k+hdmi+capture+card&qid=1691848252&sprefix=4k+hdmi+cap%2Caps%2C97&sr=8-2-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&psc=1
also got 2 MS2109 on way 
sk6812


any help would be appreciated. alot. 

ive had lytmni system they ended up sending me 3 sets of lights and 2 tv boxes. and currently got philps system in.

**Reply by awawa-dev** [reply]
Adafruit Scorpio has a dedicated firmware you can use: https://github.com/awawa-dev/HyperSerialPico#how-to-flash-it 
Officially not released yet but you can use `Adafruit_Feather_RP2040_Scorpio_prerelease.zip` archive (EXTRACT IT) from https://github.com/awawa-dev/HyperSerialPico/releases
Then you need to connect Adafruit Scorpio to Rpi using standard USB connection: that's all.

**Reply by games133** [reply]
you mean usb to usb c? and that will tell the adafruit what to do with the lights?

**Reply by games133** [reply]
also would the pi power the adafruit througth the usb? or do i need as septare power?

**Reply by awawa-dev** [reply]
Rpi running HyperHDR (Adalight driver configured as described in HyperSerialPico manual) => serial port connection (USB) => USB to USB-C cable => HyperSerialPico running on Adafruit Scorpio => GPIO specified in the HyperSerialPico manual to LEDs
You also need a common ground between LEDs and Adafruit board.
Rpi power the Adafruit Scorpio througth the usb.

**Reply by games133** [reply]
so ill be able to pin off the adafruit borad and light and raspberry together with a compact wire connector.

**Reply by awawa-dev** [reply]
I don't quite understand what you want to do

**Reply by games133** [reply]
wire connector :- https://www.amazon.co.uk/connectors-Splicing-Connector-Electrical-Connectors/dp/B0BKY43TYL/ref=sr_1_3_sspa?keywords=electrical%2Bwire%2Bconnectors&qid=1691855070&sr=8-3-spons&sp_csd=d2lkZ2V0TmFtZT1zcF9hdGY&th=1

would the light work with out a capture card or HDMI splitter for test purposes? 

cant seam get any whear lights dont seam light up at all

ok got some whear there a fault with the adafruit driver. hyper not picking it up for some reason. eben when nothink else is connected.

**Reply by awawa-dev** [reply]
Did you set up the driver correctly in HyperHDR? Instead of USB grabber use effects in the remote tabs to test it. Upload to pastebin **full** logs taken 1-2 minutes after HyperHDR restart

**Reply by games133** [reply]
i put the adafruit into program mode as your instruction. my pc picked it up and i dropped the file in. ive got sk6812 rgbw  natural white. 
so picked the sk6812neutral. then computer made sound as it was disconected.  then connected it up to my pi tryed reeboot and it dont seam pic it up
yea tried effects nowt happend.

hear a paste bin link i think it might work
https://pastebin.com/J1QtEJWq

**Reply by awawa-dev** [reply]
Are you sure you have selected proper COM port from the list? Did you try the others?
```
2023-08-12T19:44:26.310Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:228) productIdentifier: 0x0
2023-08-12T19:44:26.310Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:229) vendorIdentifier:  0x0
2023-08-12T19:44:26.310Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:230) serialNumber:      
2023-08-12T19:44:27.613Z [LEDDEVICE_ADALIGHT] Could not detect HyperSerialEsp8266/HyperSerialESP32 device
```
On Rpi AMA0 is usually (not always) the bluetooth device: https://github.com/awawa-dev/HyperHDR/wiki/Quick-start#configuring-the-led-strip Disconnect everything from Adafruit (leave only USB) and test again. If it's not detected use PC to check if it is detected as a serial port (in the Windows device manager panel). Which exactly file (full filename) did you upload to Adafruit? Extracted from Adafruit_Feather_RP2040_Scorpio_prerelease.zip archive? Try to upload it again. 

Do use use the same USB cable for PC and Rpi?

**Reply by games133** [reply]
that would explaine why i get nowt. there isnt another driver to select. the file i used is that is one. ive tried reloading it and it dont work. dont think my pc pics it up as anything. it will go into duf mode but no light come on or anythig. could i of damage it when soildering the pins? i bought 2 and both are the same. tried differ usb cables as well. the file i used is that is one.
HyperSerialPico_sk6812Neutral.uf2

ok after a couple tries my computer picks it up as a usb serial device (com5)
but once disconcerted
 id dont pic it up and more

**Reply by awawa-dev** [reply]
Maybe the cable is broken? Your PC should detect it (as serial port) every time it's connected.

**Reply by awawa-dev** [reply]
You can also try standard firmware, not from `Adafruit_Feather_RP2040_Scorpio_prerelease.zip` but regular one: https://github.com/awawa-dev/HyperSerialPico/releases/download/v3.0.0.0/HyperSerialPico_sk6812Neutral.uf2 It won't work with Adafruit Scorpio GPIO but just to test if it will be discovered by Windows PC as a serial port every time you connect it.

**Reply by games133** [reply]
ive managed to get hyper to pic it up but had to load the file on it through rpi. hyper has picked it up as pico (/dev/ttyacmo).

and now ive got a dancing set lights. 

just need try a reboot of the rpi and see if it still works
 
so after testing the lights. i decided turn the power off and found the adafruit not to work again

just wondering if you had any experience like this with adafruit

**Reply by awawa-dev** [reply]
If the correct firmware is uploaded, it will always work. Even now I checked several times my Adafruit Scorpio in which the firmware was uploaded more than 2 weeks ago and it is detected every time by PC. Anyway, this mechanism is not unique to Scorpio, it's just how every rp2040-based board works and this chip stores the firmware.

**Reply by awawa-dev** [reply]
Maybe you have a bad batch of rp2040? https://forums.raspberrypi.com/viewtopic.php?t=333918

**Reply by games133** [reply]
So how do I add 

#include "hardware/xosc.h"
stdio_init_all();
xosc_init(); // #define PICO_XOSC_STARTUP_DELAY_MULTIPLIER 64
 In to the file I need?

**Comment by awawa-dev** [comment]
I compiled a new firmware for your Adafruit with a workaround for the broken/unstable oscillator. Use this Artifacts: https://github.com/awawa-dev/HyperSerialPico/actions/runs/5843994589
Inside Artifacts you will find new `Adafruit_Feather_RP2040_Scorpio.zip` that contains `HyperSerialPico_sk6812Neutral.uf2`. Let me know if it helps.

**Reply by games133** [reply]
hi that worked straight away every time i coinnect and disconet from my pc it shows up. put it on both board and both have same out come. 

your a super star

**Reply by awawa-dev** [reply]
Thank you for your feedback. It's an important finding that some rp2040's may not be stable on startup. Today I will release a new version of HyperSerialPico and the firmware for Adafruits will receive this fix.

**Reply by games133** [reply]
ive tested both my boards and both work great. ive tested my kit connected to my tv and the lights seam adjust and move in sync
but i have notived 2 error when looking at the logs. i could set up a bin for them and you look when ever or post the lines on hear. 

im happy to help as and when i can. im no programmer but dont mind playing with my system buying things to try

**Reply by awawa-dev** [reply]
Some errors can be ignored, depending on the situation. What does the log say?

**Reply by games133** [reply]
cant rember now ive took it apart and waiting for new tv or figure how im going mount the light.  also waiting for a box to come to put it all in. 
but i can put it all back together and get the log file if you want skim over it. 
i am using alternative capture card thoe.

**Comment by games133** [comment]
if i wanted to drive 2 sets of lights would i still use pin 0 & 1? and do i have to change that?


the 2 lines of code im not to sure are problem. 
2023-08-13T13:38:07.420Z [LEDDEVICE_ADALIGHT] Detected Rp2040 type board. HyperHDR skips the reset. State: 1, 1

2023-08-13T19:37:34.485Z [LEDDEVICE_ADALIGHT] The refresh timer is enabled ('Refresh time' > 0) and may limit the performance of the LED driver. Ignore this error if you set it on purpose for some reason (but you almost never need it).

**Reply by awawa-dev** [reply]
That could be a real problem: 
`2023-08-13T19:37:34.485Z [LEDDEVICE_ADALIGHT] The refresh timer is enabled ('Refresh time' > 0) `
set refresh time to 0 in the LED device driver. BTW, how high did you set the refresh rate in the smoothing unit (and how many LEDs do you have)?

**Reply by games133** [reply]
ive not set the smoothing unit yet i dont think lol. im only playing around at moment and set leds to 265. its roughly what my current tv size and new tv size is but not exact due not been mount yet

**Reply by awawa-dev** [reply]
OK, without smoothing if the video source (USB grabber) framerate is equal or lower then 60FPS, in the Adalight driver you should set 'refresh time' to 0 to eliminate possible & unnecessary lag.

**Reply by games133** [reply]
ive not connected up to a tv yet still awaiting part from chine to replace expensive ones from amazon. 

had a thought today at worl. if you had the pi and power block and that some wear could you use a long usb-c lead and extend teh wires for power to the light strip so the light controller and light are on the tv?

**Reply by games133** [reply]
hi ive set it all up. all up and running just need make some tweaks. been trying log onto raspberrypi:8090 throught a laptop but it wont let me. even if i turn off the web browser on my pc whitch i first used to confgure hyper. is there something i need switch to be able log on with laptop?

**Reply by awawa-dev** [reply]
You can access HyperHDR from multiple sources at once, so the problem is your network. I am not sure which devices are using your Wi-Fi and which are using Ethernet. In the case of wireless communication, enabled wifi isolation can cause problems. Try using the IP address instead of the Rpi name and try pinging it from your laptop as well.
